### PR TITLE
feat: miragte from auth module v4 to v5

### LINKED ISSUE #10: Compability with new laravel's airlock feature
Hi. i want to open this thread to find a solution into this implementation.

However i succefully got authentication , in echo i receive only 
```
{"message":"Unauthenticated."}
Client can not be authenticated, got HTTP status 401
```
even if i set `Broadcast::routes(['middleware' => ['auth:airlock']]);` as suggested in documentation https://laravel.com/docs/7.x/airlock#authorizing-private-broadcast-channels.

### PR DESCRIPTION
**Pull request by JoaoPedroAS51**
This PR migrate from auth module v4 to v5. It also add referer and x-xsrf-token headers to work with laravel sanctum provider.

Closes #10

### CODE REVIEW

**Comment by ricardogobbosouza** [comment]
Hi @JoaoPedroAS51 
It looks good, can we merge?

**Comment by JoaoPedroAS51** [comment]
Hi @ricardogobbosouza! Unfortunately we can't merge it yet. We are making more refactors before release v5 and I will probably need to update this PR again :)

**Comment by iranagame** [comment]
Any update on this!?
We're using `auth-next` because of token refresh capability and we still have problem with `laravel-echo`, I have to move our app to production in few days and still didn't find any solution to support this lib as well!!!

**Comment by iranagame** [comment]
@JoaoPedroAS51 @ricardogobbosouza

**Comment by JoaoPedroAS51** [comment]
Hi @iranagame! You can use [my fork](https://github.com/JoaoPedroAS51/laravel-echo/tree/feat/auth-v5) until we merge this PR :)

Just add something like `"@nuxtjs/laravel-echo": "JoaoPedroAS51/laravel-echo#feat/auth-v5"` to your `package.json`

File: lib/plugin.js
Comment (aaronhuisinga) [inline lib/plugin.js:22]: This doesn't seem to work with the latest nuxt/auth version. Changing the line to `const token = strategy.token.get()` fixes this PR.

**Comment by aaronhuisinga** [comment]
I believe [this](https://github.com/nuxt-community/auth-module/pull/690) changed the method for getting the token from `$auth.token.get()` to `$auth.strategy.token.get()`. Due to this change, this PR is currently broken. I left a comment on the plugin.js file with a fix that I can verify works.

@JoaoPedroAS51 can you take a verify my fix is correct and update the code? We're trying to use this package with Auth v5 and cannot until this is fixed. Thanks so much!

File: lib/plugin.js
Comment (JoaoPedroAS51) [reply lib/plugin.js:22]: Yes, you're right. The token instance has been moved to strategy.

**Comment by JoaoPedroAS51** [comment]
@aaronhuisinga Done! :)

File: lib/plugin.js
Comment (JoaoPedroAS51) [reply lib/plugin.js]: Fixed in 8635bd52e128fe72f03f7f67d4e46e4649bd09fe

**Comment by JoaoPedroAS51** [comment]
@ricardogobbosouza Can this package have a beta version? That way we could merge this PR :)

**Comment by aaronhuisinga** [comment]
Thanks @JoaoPedroAS51!

**Comment by ricardogobbosouza** [comment]
Sure @JoaoPedroAS51 
I'll do it this week

File: lib/plugin.js
Comment (aaronhuisinga) [inline lib/plugin.js:11]: @JoaoPedroAS51 What's the purpose of setting the referer header here? Chrome is logging the error `Refused to set unsafe header "referer"` every time I connect to a channel via Pusher. Is this required for something?

Comment (JoaoPedroAS51) [reply lib/plugin.js:11]: If I'm not wrong, it is required for Laravel Sanctum.

Comment (aaronhuisinga) [reply lib/plugin.js:11]: I'm using Passport and it's definitely not required there. Would it make more sense to move it down to line 19 in the Sanctum conditional? The package does work properly with it here (at least in my usage), but moving it would remove the errors in the Chrome console.

Comment (JoaoPedroAS51) [reply lib/plugin.js:11]: Yeah, you're right! If it's only required for Sanctum, it should be moved to Sanctum conditional. I'll do that tomorrow :)

Comment (JoaoPedroAS51) [reply lib/plugin.js:11]: Thanks for your help reviewing and testing the code!

Comment (JoaoPedroAS51) [reply lib/plugin.js]: Fixed in 66e0111

**Comment by aaronhuisinga** [comment]
Thanks so much for all your work on this @JoaoPedroAS51 and @ricardogobbosouza!

**Comment by aaronhuisinga** [comment]
@ricardogobbosouza Any plans to get this merged into a pre-release version?

**Comment by ricardogobbosouza** [comment]
Hi @aaronhuisinga 
I'll do it until the end of January
Sorry for the delay

**Review by jalallinux — APPROVED** [review]
its good and work

**Comment by ricardogobbosouza** [comment]
Apologies for the delay
Releasing an alpha version today =)

**Pull request merged**
